src/software/ai/hl/stp/play/penalty_kick_play.cpp

```diff
@@ -14,44 +14,61 @@ std::string PenaltyKickPlay::getName() const
 
 bool PenaltyKickPlay::isApplicable(const World &world) const
 {
-    if (world.gameState().isOurPenalty())
-    {
-        return true;
-    }
-    else
-    {
-        return false;
-    }
+    return (world.gameState().isReadyState() || world.gameState().isSetupState()) &&
+           world.gameState().isOurPenalty();
 }
 
 bool PenaltyKickPlay::invariantHolds(const World &world) const
 {
-    if (world.gameState().isOurPenalty())
-    {
-        return true;
-    }
-    else
-    {
-        return false;
-    }
+    return world.gameState().isOurPenalty() && !world.gameState().isStopped() &&
+           !world.gameState().isHalted();
 }
 
 void PenaltyKickPlay::getNextTactics(TacticCoroutine::push_type &yield)
 {
     auto penalty_shot_tactic = std::make_shared<PenaltyKickTactic>(
         world.ball(), world.field(), world.enemyTeam().goalie(), true);
 
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::BALL);
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::HALF_METER_AROUND_BALL);
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::ENEMY_DEFENSE_AREA);
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+
+    auto shooter_setup_move = std::make_shared<MoveTactic>(true);
+
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::ENEMY_DEFENSE_AREA);
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::HALF_METER_AROUND_BALL);
+
     auto move_tactic_2 = std::make_shared<MoveTactic>(true);
+    move_tactic_2->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_2->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_3 = std::make_shared<MoveTactic>(true);
+    move_tactic_3->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_3->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_4 = std::make_shared<MoveTactic>(true);
+    move_tactic_4->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_4->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_5 = std::make_shared<MoveTactic>(true);
+    move_tactic_5->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_5->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_6 = std::make_shared<MoveTactic>(true);
+    move_tactic_6->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_6->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
```

Comment:
Similar comment here to my comment for PenaltyKickEnemyChanges, could we modify the avoid area to actually encompass the entire enemy half instead of whitelisting the enemy half?

Reply:
Created #980 and commented the avoid areas that will need to be removed.